kasa/device.py

```diff
@@ -369,13 +369,125 @@ def __repr__(self):
         ),
     }
 
-    def __getattr__(self, name) -> bool:
-        if name in self._deprecated_attributes:
+    _deprecated_effect_attributes = {
+        # Light Effects
+        # The return values for effect is a str instead of dict so the lightstrip
+        # modules have a _deprecated method to return the value as before.
+        "effect": (
+            Module.LightEffect,
+            lambda self: self.modules[Module.LightEffect]._deprecated_effect
+            if Module.LightEffect in self.modules
+            and hasattr(self.modules[Module.LightEffect], "_deprecated_effect")
+            else self.modules[Module.LightEffect].effect
+            if Module.LightEffect in self.modules
+            else None,
```

Comment:
I'm wondering if there's a possibility to avoid lambdas & generalize the checks among all these attributes?

Reply:
Ah, I should've mentioned that I meant this as a general comment as I saw that many of the lambdas share a very similar structure (i.e., return a function by name "x" if a given module is available).

So yes, it'd be nice to generalize at least the common case, but wouldn't something like this work in this case?
```python
def get_replacing_attr(module: Module, *attrs):
    if module not in self.modules:
        return None

    for attr in attrs:
        if (new_attr := getattr(module, attr)) is not None:
            return new_attr

    return None

get_replacing_attr(Module.LightEffect, "_deprecated_effect", "effect", "something else?")
```

Reply:
Thanks for the suggestion, I've applied that and it looks much cleaner now.